### PR TITLE
fix(nodejs): validate sendAndWait timeout to fix malformed error message

### PR DESCRIPTION
**Pull request by gimenete**
## Summary

Fixes the malformed timeout error reported in #915.

When `session.sendAndWait()` received a non-numeric `timeout` (e.g. an object or `NaN`), the value was passed straight to `setTimeout()`, which coerces an invalid delay to **0ms**. The session then rejected almost immediately with a misleading message:

> `Timeout after [object Object]ms waiting for session.idle`

instead of surfacing the real problem (an invalid argument).

This PR validates the `timeout` argument up front and throws a clear `TypeError` when it is not a non-negative finite number of milliseconds:

> `sendAndWait timeout must be a non-negative number of milliseconds (got object)`

This also fixes the secondary symptom of a **spurious immediate timeout** caused by the `NaN`/object → `0ms` coercion.

## Changes

- `nodejs/src/session.ts`: validate `timeout` in `sendAndWait()` before use; document the new `@throws` in the JSDoc.
- `nodejs/test/client.test.ts`: add regression tests covering object, `NaN`, and negative timeouts, asserting the error message no longer contains `[object Object]`.

## Scope note

#915 also reports that `resumeSession()` with `customAgents` never reaches `idle`. The SDK already forwards `customAgents` in the `session.resume` payload, so that behavior is on the CLI runtime side and is not addressable in this SDK. This PR therefore targets only the malformed-message bug.

## Testing

From `nodejs/`:
- `npx vitest run test/client.test.ts -t "sendAndWait timeout validation"` (fails before the fix, passes after)
- `npx vitest run test/client.test.ts` — all 111 pass
- `npm run lint` — 0 errors
- `npm run format:check` — clean
- `npm run typecheck` — clean

### CODE REVIEW

File: nodejs/test/client.test.ts [hunk 1]
```diff
@@ -24,6 +24,38 @@ describe("CopilotClient", () => {
...
+        // which coerced it to a 0ms delay and rejected with a malformed
+        // "Timeout after [object Object]ms" message instead of a clear error.
+        it("rejects with a TypeError mentioning the received timeout type", async () => {
+            const session = new CopilotSession("session-1", {} as any);
+            await expect(session.sendAndWait("hello", { ms: 30000 } as any)).rejects.toThrow(
+                TypeError
+            );
+            await expect(session.sendAndWait("hello", { ms: 30000 } as any)).rejects.toThrowError(
+                /sendAndWait timeout must be a non-negative number of milliseconds/
+            );
```
[the remark this answers is not in the document]
Comment (gimenete) [reply nodejs/test/client.test.ts]: Good call. Refactored to capture the rejection once via a small `captureRejection` helper and assert both the `TypeError` type and the exact message against a single invocation. Fixed in 30dc7e9.

File: nodejs/test/client.test.ts [hunk 2]
```diff
@@ -24,6 +24,38 @@ describe("CopilotClient", () => {
...
+        });
+
+        it("does not produce an [object Object] message for an object timeout", async () => {
+            const session = new CopilotSession("session-1", {} as any);
+            await expect(session.sendAndWait("hello", { ms: 30000 } as any)).rejects.toThrowError(
+                /timeout must be a non-negative number of milliseconds \(got object\)/
+            );
```
[the remark this answers is not in the document]
Comment (gimenete) [reply nodejs/test/client.test.ts]: Tightened all the assertions to match the full message (including the `sendAndWait` prefix and the exact `(got ...)` suffix) instead of a generic substring. Fixed in 30dc7e9.

File: nodejs/src/session.ts [hunk 3]
```diff
@@ -275,6 +276,21 @@ export class CopilotSession {
...
+        // immediate timeout with a malformed "Timeout after [object Object]ms"
+        // message instead of a clear, actionable error.
+        if (
+            timeout !== undefined &&
+            (typeof timeout !== "number" || !Number.isFinite(timeout) || timeout < 0)
+        ) {
+            const received = typeof timeout === "number" ? `${timeout}` : typeof timeout;
+            throw new TypeError(
+                `sendAndWait timeout must be a non-negative number of milliseconds (got ${received})`
+            );
```
[the remark this answers is not in the document]
Comment (gimenete) [reply nodejs/src/session.ts]: Agreed — `typeof null === "object"` made that misleading. Special-cased `null` so it now reports `(got null)`, and added a regression test for it. Fixed in 30dc7e9.

**Comment by SteveSandersonMS** [comment]
Thanks for submitting this, @gimenete. Having looked into it, I don't think this is correct:

1. We rely on TypeScript checking types. We don't do runtime checks for arguments of the wrong type across all APIs. If someone chooses to use Copilot SDK with JavaScript instead of TypeScript that's fine but it becomes their responsibility to pass args of the right type.
2. Even if we did add the type checks proposed in this PR, it wouldn't change the repro in #915 to start working, because this PR just exchanges one exception with a different exception. It would still fail.

The only edge case we might check for is negative number values (since requiring the timeout to be positive isn't expressed in the type declaration), but at runtime a negative timeout value is allowed so it's not really a problem.
